### ISSUE TITLE
 java.io.NotSerializableException: io.hyperfoil.core.steps.ScheduleDelayStep$Builder

### ISSUE THREAD

**Issue by willr3** [Hyperfoil/Hyperfoil#63]
```
[hyperfoil@in-vm]$ upload examples/eshop.hf.yaml
Loaded benchmark eshop, uploading...
java.lang.RuntimeException: java.io.NotSerializableException: io.hyperfoil.core.steps.ScheduleDelayStep$Builder
```

**Comment by willr3** [comment]
I built from master 7f831552

**Comment by rvansa** [comment]
My apologies, I've fixed it. Some lambdas generated by the builder captured the `this` reference.
I am bit surprised that despite the lambda is `Serializable` it allowed me to create it with this non-serializable object (`Serializable lambda = foo -> this;`).

It would be great to make an automatic test for all possible builder products as this happens quite often, but enumerating all the possibilities is complicated. I've added a check that all examples are serializable.

**Comment by willr3** [comment]
What about running the examples in CI?

**Comment by rvansa** [comment]
Yes, that would certainly help to reveal some problems early on; the serializability of examples is already checked, though. What we'd really need is having a test for each step/processor/xxx that would try it with each and every sensible combination of properties. And another test to verify that each step/xxx has the test defined, ofc.

Closing this issue as fixed in f2b70550f089b3dbf47b596ac58b98e61bc43b5d
